### PR TITLE
Feature: Added a theming option to display a background image

### LINKED ISSUE #12636: Feature: Add support for setting a background image for the app
### What feature or improvement do you think would benefit Files?

Hiya, please add option in the Appearance section of the settings to add custom image as background.

### Requirements

- Add combo box with sub options "Image" and "None"
- Include sub setting to select image source
- Include sub setting to change transparency
- Include sub setting to choose image placement

### Files Version

2.5.1.0

### Windows Version

Windows 11 (22H2): 22621

### Comments

_No response_

### PR DESCRIPTION
**Pull request by yaira2**
**Resolved / Related Issues**
- [x] Were these changes approved in an issue or discussion with the project maintainers? In order to prevent extra work, feature requests and changes to the codebase must be approved before the pull request will be reviewed. This prevents extra work for the contributors and maintainers.
   Closes #12636

**Validation**
How did you test these changes?
- [x] Did you build the app and test your changes?
- [ ] Did you check for accessibility? You can use Accessibility Insights for this.
- [ ] Did you remove any strings from the en-us resource file?
   - [ ] Did you search the solution to see if the string is still being used? 
- [ ] Did you implement any design changes to an existing feature?
   - [ ] Was this change approved?
- [x] Are there any other steps that were used to validate these changes?
   1. Open settings file and assign path to an image file.
   2. Open Files and confirm the image is displayed.
   3. Adjust the different properties in the settings UI, and confirm the different combinations work properly.

**ToDo**
- [ ] Add UI for selecting image
- [ ] Add UI for removing existing image (ie. setting the path value to null)

**Screenshots (optional)**
image
image
image

### CODE REVIEW

File: src/Files.App/Services/Settings/IAppearanceSettingsService.cs [hunk 1]
```diff
@@ -53,5 +56,30 @@ public interface IAppearanceSettingsService : IBaseSettingsService, INotifyPrope
 		/// Gets or sets a value for the theme system backdrop.
 		/// </summary>
 		BackdropMaterialType AppThemeBackdropMaterial { get; set; }
+
+		/// <summary>
+		/// Gets or sets a value for the app background image source
+		/// </summary>
+		string AppBackgroundImageSource { get; set; }
```
Comment (0x5bfa) [inline src/Files.App/Services/Settings/IAppearanceSettingsService.cs:63]: Following others,
AppThemeBackgroundImageSource sounds good.

Others below too.

File: src/Files.App/ViewModels/MainPageViewModel.cs [hunk 2]
```diff
@@ -57,6 +60,25 @@ public bool ShouldPreviewPaneBeDisplayed
 			set => SetProperty(ref shouldPreviewPaneBeDisplayed, value);
 		}
 
+		public Stretch AppBackgroundImageFit => AppearanceSettingsService.AppBackgroundImageFit;
```
Comment (0x5bfa) [inline src/Files.App/ViewModels/MainPageViewModel.cs:63]:
```suggestion
		public Stretch AppBackgroundImageFit
			=> AppearanceSettingsService.AppBackgroundImageFit;
```

Other below too

File: src/Files.App/ViewModels/MainPageViewModel.cs [hunk 3]
```diff
@@ -57,6 +60,25 @@ public bool ShouldPreviewPaneBeDisplayed
...
+
+		public ImageSource? AppBackgroundImageSource
+		{
+			get
+			{
+				return string.IsNullOrEmpty(AppearanceSettingsService.AppBackgroundImageSource)
+					? null
+					: new BitmapImage(new Uri(AppearanceSettingsService.AppBackgroundImageSource, UriKind.RelativeOrAbsolute));
+			}
+		}
```
Comment (0x5bfa) [inline src/Files.App/ViewModels/MainPageViewModel.cs:75]:
```suggestion
		public ImageSource? AppBackgroundImageSource =>
			string.IsNullOrEmpty(AppearanceSettingsService.AppBackgroundImageSource)
				? null
				: new BitmapImage(new Uri(AppearanceSettingsService.AppBackgroundImageSource, UriKind.RelativeOrAbsolute));
```

**Review by 0x5bfa — COMMENTED** [review]
As of UI, why not put Browse button in the expander header and put Remove button as a normal button in the inner child showing path?

File: src/Files.App/Services/Settings/IAppearanceSettingsService.cs (hunk 1 above)
Comment (yaira2) [reply src/Files.App/Services/Settings/IAppearanceSettingsService.cs]: Resolved in the latest commit

File: src/Files.App/ViewModels/MainPageViewModel.cs (hunk 3 above)
Comment (yaira2) [reply src/Files.App/ViewModels/MainPageViewModel.cs]: Resolved in the latest commit

File: src/Files.App/ViewModels/MainPageViewModel.cs (hunk 2 above)
Comment (yaira2) [reply src/Files.App/ViewModels/MainPageViewModel.cs]: Resolved in the latest commit

**Review by 0x5bfa — APPROVED** [review]
LGTM code wise

File: src/Files.App/ViewModels/Settings/AppearanceViewModel.cs [hunk 4]
```diff
@@ -42,7 +56,64 @@ public AppearanceViewModel(IUserSettingsService userSettingsService, IResourcesS
...
+			ImageVerticalAlignmentTypes.Add(VerticalAlignment.Bottom, "Bottom".GetLocalizedResource());
+			ImageVerticalAlignmentTypes.Add(VerticalAlignment.Stretch, "Stretch".GetLocalizedResource());
+			SelectedImageVerticalAlignmentType = ImageVerticalAlignmentTypes[UserSettingsService.AppearanceSettingsService.AppThemeBackgroundImageVerticalAlignment];
+
+			// HorizontalAlignment
+			ImageHorizontalAlignmentTypes.Add(HorizontalAlignment.Left, "Top".GetLocalizedResource());
+			ImageHorizontalAlignmentTypes.Add(HorizontalAlignment.Center, "Center".GetLocalizedResource());
+			ImageHorizontalAlignmentTypes.Add(HorizontalAlignment.Right, "Bottom".GetLocalizedResource());
+			ImageHorizontalAlignmentTypes.Add(HorizontalAlignment.Stretch, "Stretch".GetLocalizedResource());
+			SelectedImageHorizontalAlignmentType = ImageHorizontalAlignmentTypes[UserSettingsService.AppearanceSettingsService.AppThemeBackgroundImageHorizontalAlignment];
```
Comment (hishitetsu) [inline src/Files.App/ViewModels/Settings/AppearanceViewModel.cs:81]: * Shouldn't horizontal alignment be left, center, or right?
* Does `Stretch` make sense?

**Review by hishitetsu — COMMENTED** [review]
Sorry for the late review.

File: src/Files.App/ViewModels/Settings/AppearanceViewModel.cs (hunk 4 above)
Comment (yaira2) [reply src/Files.App/ViewModels/Settings/AppearanceViewModel.cs:81]: I just went with all the available options, but we can remove stretch if it doesn't make a difference.

Comment (hishitetsu) [reply src/Files.App/ViewModels/Settings/AppearanceViewModel.cs:81]: Then we should remove stretch.

**Pull request merged**